gc/verbose/VerboseHandlerOutput.cpp

```diff
@@ -225,110 +234,48 @@ MM_VerboseHandlerOutput::handleInitializedInnerStanzas(J9HookInterface** hook, u
 }
 
 void
-MM_VerboseHandlerOutput::handleInitializedRegion(J9HookInterface** hook, uintptr_t eventNum, void* eventData)
+MM_VerboseHandlerOutput::outputInitializedRegion(MM_EnvironmentBase *env, MM_VerboseBuffer *buffer)
 {
-	MM_InitializedEvent* event = (MM_InitializedEvent*)eventData;
-	MM_VerboseWriterChain* writer = _manager->getWriterChain();
-	MM_EnvironmentBase* env = MM_EnvironmentBase::getEnvironment(event->currentThread);
+	MM_GCExtensionsBase *extension = env->getExtensions();
+	OMR_VM *omrVM = env->getOmrVM();
 #if defined(OMR_GC_DOUBLE_MAP_ARRAYLETS)
 	bool isArrayletDoubleMapRequested = _extensions->isArrayletDoubleMapRequested;
 	const char *arrayletDoubleMappingStatus = _extensions->indexableObjectModel.isDoubleMappingEnabled() ? "enabled" : "disabled";
 	const char *arrayletDoubleMappingRequested = isArrayletDoubleMapRequested ? "true" : "false";
 #endif /* OMR_GC_DOUBLE_MAP_ARRAYLETS */
-
-	writer->formatAndOutput(env, 1, "<region>");
-	writer->formatAndOutput(env, 2, "<attribute name=\"regionSize\" value=\"%zu\" />", event->regionSize);
-	writer->formatAndOutput(env, 2, "<attribute name=\"regionCount\" value=\"%zu\" />", event->regionCount);
-	writer->formatAndOutput(env, 2, "<attribute name=\"arrayletLeafSize\" value=\"%zu\" />", event->arrayletLeafSize);
+	buffer->formatAndOutput(env, 1, "<region>");
+	buffer->formatAndOutput(env, 2, "<attribute name=\"regionSize\" value=\"%zu\" />", extension->getHeap()->getHeapRegionManager()->getRegionSize());
+	buffer->formatAndOutput(env, 2, "<attribute name=\"regionCount\" value=\"%zu\" />", extension->getHeap()->getHeapRegionManager()->getTableRegionCount());
+	buffer->formatAndOutput(env, 2, "<attribute name=\"arrayletLeafSize\" value=\"%zu\" />", omrVM->_arrayletLeafSize);
 #if defined(OMR_GC_DOUBLE_MAP_ARRAYLETS)
 	if (_extensions->isVLHGC()) {
-		writer->formatAndOutput(env, 2, "<attribute name=\"arrayletDoubleMappingRequested\" value=\"%s\"/>", arrayletDoubleMappingRequested);
+		buffer->formatAndOutput(env, 2, "<attribute name=\"arrayletDoubleMappingRequested\" value=\"%s\"/>", arrayletDoubleMappingRequested);
 		if (isArrayletDoubleMapRequested) {
-			writer->formatAndOutput(env, 2, "<attribute name=\"arrayletDoubleMapping\" value=\"%s\"/>", arrayletDoubleMappingStatus);
+			buffer->formatAndOutput(env, 2, "<attribute name=\"arrayletDoubleMapping\" value=\"%s\"/>", arrayletDoubleMappingStatus);
 		}
 	}
 #endif /* OMR_GC_DOUBLE_MAP_ARRAYLETS */
-	writer->formatAndOutput(env, 1, "</region>");
+	buffer->formatAndOutput(env, 1, "</region>");
 }
 
 void
-MM_VerboseHandlerOutput::handleInitialized(J9HookInterface** hook, uintptr_t eventNum, void* eventData)
+MM_VerboseHandlerOutput::handleInitializedRegion(J9HookInterface** hook, uintptr_t eventNum, void* eventData)
```

Comment:
Is this needed, can it be deleted now that we have `outputInitializedRegion`? The call sites can directly call `outputInitializedRegion`. I think it only ever called through `handleInitialized` and `openFile`

Reply:
Similarly, `handleInitializedInnerStanzas ` can be deleted now that you have the `outputInitializedInnerStanzas` variant